backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -12,54 +12,157 @@
 
 public class DocxGenerator extends OutputGenerator<byte[]> {
 
+    private static final String UNKNOWN_LICENSE_NAME = "Unknown license name";
+
     public DocxGenerator() {
         super("docx", "License information as DOCX", true, "application/vnd.openxmlformats-officedocument.wordprocessingml.document");
     }
 
     @Override
-    public byte[] generateOutputFile(Collection<LicenseInfoParsingResult> projectLicenseInfoResults, String projectName, String licenseInfoHeaderText) throws SW360Exception {
+    public byte[] generateOutputFile(Collection<LicenseInfoParsingResult> projectLicenseInfoResults, String projectName, String projectVersion, String licenseInfoHeaderText) throws SW360Exception {
         try {
             XWPFDocument document = new XWPFDocument(this.getClass().getResourceAsStream("/templateFrontpageContent.docx"));
-            fillDocument(document, projectLicenseInfoResults, projectName, licenseInfoHeaderText);
-
+            fillDocument(document, projectLicenseInfoResults, projectName, projectVersion, licenseInfoHeaderText);
             ByteArrayOutputStream docxOutputStream = new ByteArrayOutputStream();
             document.write(docxOutputStream);
             docxOutputStream.close();
             return docxOutputStream.toByteArray();
+        } catch (XmlException e) {
+            throw new SW360Exception("Got XmlException while generating docx document: " + e.getMessage());
         } catch (IOException ioe) {
             throw new SW360Exception("Got IOException when generating docx document: " + ioe.getMessage());
         }
     }
 
-    private void fillDocument(XWPFDocument document, Collection<LicenseInfoParsingResult> projectLicenseInfoResults, String projectName, String licenseInfoHeaderText) throws IOException {
+    private void fillDocument(XWPFDocument document, Collection<LicenseInfoParsingResult> projectLicenseInfoResults,
+                              String projectName, String projectVersion, String licenseInfoHeaderText) throws XmlException {
+        replaceText(document, "$license-info-header", licenseInfoHeaderText);
+        replaceText(document, "$project-name", projectName);
+        replaceText(document, "$project-version", projectVersion);
+
+        fillReleaseBulletList(document, projectLicenseInfoResults);
+        fillReleaseDetailList(document, projectLicenseInfoResults);
+        fillLicenseList(document, projectLicenseInfoResults);
+    }
+
+    private void fillReleaseBulletList(XWPFDocument document, Collection<LicenseInfoParsingResult> projectLicenseInfoResults) throws XmlException {
+        List<String> releaseList = new ArrayList<>();
+        for (LicenseInfoParsingResult result : projectLicenseInfoResults) {
+            releaseList.add(nullToEmptyString(result.getName()) + " " + nullToEmptyString(result.getVersion()));
+        }
+        addBulletList(document, releaseList, true);
+        addPageBreak(document.createParagraph().createRun());
```

Comment:
Here I would expect a util function `addPageBreak(dokument)`. Same for `addNewLines(document.createParagraph().createRun(), 1)` replaced by `addNewLines(document, 1)`

Reply:
Changed parameters of 'addPageBreak' method to XWPFDocument instead of XWPFRun.
--> addPageBreak(document);
Added overload method 'addNewLines' with parameter XWPFDocument
--> addNewLines(document, 1);